src/main/java/org/folio/service/caches/MappingParametersCache.java

```diff
@@ -0,0 +1,75 @@
+package org.folio.service.caches;
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.Json;
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.folio.OrganizationCollection;
+import org.folio.processing.mapping.defaultmapper.processor.parameters.MappingParameters;
+import org.folio.rest.util.OkapiConnectionParams;
+import org.folio.rest.util.RestUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class MappingParametersCache {
+  private static final Logger LOGGER = LogManager.getLogger();
+  public static final String ORGANIZATIONS = "/organizations/organizations";
+
+  @Value("${orders.cache.organization.expiration.seconds:3600}")
+  private long cacheExpirationTime;
+
+  private AsyncCache<String, MappingParameters> cache;
+
+  @Autowired
+  public MappingParametersCache(Vertx vertx) {
+    cache = Caffeine.newBuilder()
+      .expireAfterAccess(cacheExpirationTime, TimeUnit.SECONDS)
+      .executor(task -> vertx.runOnContext(v -> task.run()))
+      .buildAsync();
+  }
+
+  public Future<MappingParameters> get(OkapiConnectionParams params) {
+    try {
+      return Future.fromCompletionStage(cache.get(params.getTenantId(), (key, executor) -> loadJobProfileSnapshot(params)));
+    } catch (Exception e) {
+      LOGGER.warn("get:: Error loading organizations from cache, tenantId: '{}'", params.getTenantId(), e);
+      return Future.failedFuture(e);
+    }
+  }
+
+  private CompletableFuture<MappingParameters> loadJobProfileSnapshot(OkapiConnectionParams params) {
```

Comment:
propose "loadMappingParameters" or something like that

Reply:
Thanks, done